PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:5 + add:0, change matches:
-    _inherit = ['hr.contract']
-    _inherit = ['hr.employee']
-    _inherit = ['hr.work.entry']
-    _inherit = ['hr.work.entry.type']
-    _inherit = ['resource.calendar']

Total Changes: 10

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_work_entry_contract/models/hr_contract.py
+++ b/addons/hr_work_entry_contract/models/hr_contract.py
@@ -18,7 +18,7 @@ from .hr_work_intervals import WorkIntervals
 
 
 class HrContract(models.Model):
-    _inherit = ['hr.contract']
+    _inherit = 'hr.contract'
     _description = 'Employee Contract'
 
     date_generated_from = fields.Datetime(string='Generated From', readonly=True, required=True,

--- a/addons/hr_work_entry_contract/models/hr_employee.py
+++ b/addons/hr_work_entry_contract/models/hr_employee.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class HrEmployee(models.Model):
-    _inherit = ['hr.employee']
+    _inherit = 'hr.employee'
 
     def generate_work_entries(self, date_start, date_stop, force=False):
         date_start = fields.Date.to_date(date_start)

--- a/addons/hr_work_entry_contract/models/hr_work_entry.py
+++ b/addons/hr_work_entry_contract/models/hr_work_entry.py
@@ -12,7 +12,7 @@ from odoo.addons.hr_work_entry_contract.models.hr_work_intervals import WorkInte
 
 
 class HrWorkEntry(models.Model):
-    _inherit = ['hr.work.entry']
+    _inherit = 'hr.work.entry'
 
     contract_id = fields.Many2one('hr.contract', string="Contract", required=True)
     employee_id = fields.Many2one(domain=[('contract_ids.state', 'in', ('open', 'pending'))])
@@ -186,7 +186,7 @@ class HrWorkEntry(models.Model):
 
 
 class HrWorkEntryType(models.Model):
-    _inherit = ['hr.work.entry.type']
+    _inherit = 'hr.work.entry.type'
     _description = 'HR Work Entry Type'
 
     is_leave = fields.Boolean(

--- a/addons/hr_work_entry_contract/models/resource_calendar.py
+++ b/addons/hr_work_entry_contract/models/resource_calendar.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class ResourceCalendar(models.Model):
-    _inherit = ['resource.calendar']
+    _inherit = 'resource.calendar'
 
     def _get_global_attendances(self):
         return super()._get_global_attendances().filtered(lambda a: not a.work_entry_type_id.is_leave)
